feat: add vegeta http load testing document. (#2519)
tldr-pages/tldr: pages/common/vegeta.md

# vegeta

> Vegeta is a versatile HTTP load testing tool built out of a need to drill HTTP services with a constant request rate. It can be used both as a command line utility and a library.

- Launch an attack lasting 30 seconds:

> `echo "GET http://{{adress/to/the/resource}}" | vegeta attack -duration=30s`

Reviewer: `{{GET https://example.com}}` to match with other pages.

- Launch an attack and display for a report:

> `echo "GET http://{{adress/to/the/resource}}" | vegeta attack -duration=30s | vegeta report`

Reviewer: 30s, 10 in tokens

- Launch an attack and plot the results on a graph (latency against time):

> `echo "GET http://{{adress/to/the/resource}}" | vegeta attack -rate=10 -duration=30s | vegeta plot > {path/to/output}/vegeta-results.html`

Reviewer: Why not `{{path/to/results.html}}` ?

# vegeta

> HTTP load testing tool.

> > Vegeta is a versatile HTTP load testing tool built out of a need to drill HTTP services with a constant request rate. It can be used both as a command line utility and a library.

Reviewer 2: Perhaps we can make this more concise? We've mentioned that it's a load-testing tool in the first line of the description.

Reviewer 3: @antoine-amara @sbrl how about:
```suggestion
> Vegeta is both a command line utility and a library for HTTP load testing.
> See also `ab`.
```

Reviewer 2: Sounds good to me! Maybe cut the `Vegeta is` from the beginning though, as the program name is literally the page title :P

> - Launch an attack during 30 seconds:

Reviewer 2: Perhaps `lasting` would read better than `during` here?

Author: I choose to delete the first line. The second line is the official description from the vegeta repo and I prefer to take this one.

`echo "GET http://{{adress/to/the/resource}}" | vegeta attack -duration=30s`

> - Launch an attack and ask for a report:

Reviewer 2: Perhaps `display` would read better than `ask` here?

`echo "GET http://{{adress/to/the/resource}}" | vegeta attack -duration=30s | vegeta report`

> - Launch an attack and plot the results (latency/time):

Reviewer 2: ```suggestion
- Launch an attack and plot the results on a graph (latency against time):
```

Perhaps this will read better?

`echo "GET http://{{adress/to/the/resource}}" | vegeta attack -rate=10 -duration=30s | vegeta plot > {path/to/output}/vegeta-results.html`

> - Launch multiple URL attack with a file input:

Reviewer 2: ```suggestion
- Launch an attack against multiple URLs from a file:
```

Perhaps this will read better?

Also, could `requests.txt` be something like `{{path/to/request_urls.txt}}` please?

# vegeta

> Vegeta is both a command line utility and a library for HTTP load testing.
> See also `ab`.

- Launch an attack lasting 30 seconds:

> `echo "{{GET https://example.com}}" | vegeta attack -cert {{path/to/cert.pem}} -key {{path/to/key.pem}} -duration={{30s}}`

Reviewer: I am trying to understand why would I need a cert and key to communicate with example.com ? I should already have the CA installed in my machine. So unless the server uses a self-signed cert, I shouldn't have to use any certs or keys.

- Launch an attack on a server with a self-signed https certificate:

`echo "{{GET https://example.com}}" | vegeta attack -insecure -duration={{30s}}`

- Launch an attack and display for a report:

`echo "{{GET https://example.com}}" | vegeta attack -cert {{path/to/cert.pem}} -key {{path/to/key.pem}} -duration={{30s}} | vegeta report`

- Launch an attack and plot the results on a graph (latency against time):

> `echo "{{GET https://example.com}}" | vegeta attack -cert {{path/to/cert.pem}} -key {{path/to/key.pem}} -rate={{10}} -duration={{30s}} | vegeta plot > {{path/to/results.html}}`

Reviewer: Please explain what `rate` is in the description.
